lib/instrument/transformer/nskeyed.js

```diff
@@ -383,30 +482,48 @@ class Archive {
   }
 
   encodeArray (objs, archiveObj) {
-    archiveObj.$class = this.uidForArchiver('NSArray');
+    archiveObj.$class = this.uidForArchiver('NSArray', NS_OBJECT);
     archiveObj['NS.objects'] = objs.map(this.archive.bind(this));
   }
 
   encodeSet (objs, archiveObj) {
-    archiveObj.$class = this.uidForArchiver('NSSet');
+    archiveObj.$class = this.uidForArchiver('NSSet', NS_OBJECT);
     archiveObj['NS.objects'] = objs.map(this.archive.bind(this));
   }
 
   encodeDict (obj, archiveObj) {
-    archiveObj.$class = this.uidForArchiver('NSDictionary');
     archiveObj['NS.keys'] = _.keys(obj).map(this.archive.bind(this));
     archiveObj['NS.objects'] = _.values(obj).map(this.archive.bind(this));
+    //dict has to make classes after value
+    archiveObj.$class = this.uidForArchiver('NSDictionary', NS_OBJECT);
+  }
+
+  /**
+   *
+   * @param {XCTestConfiguration} obj
+   * @param {*} archiveObj
+   */
+  encodeXCTestConfiguration(obj, archiveObj) {
+    archiveObj.$class = this.uidForArchiver('XCTestConfiguration', NS_OBJECT);
+    _.entries(obj._data).forEach((entry) => {
```

Comment:
it is probably matter of taste, but I find a normal for loop more readable than forEach:

```
for (const [key, value] of _.toPairs(obj._data)) {
  ....
}
```